### PR TITLE
Cortex XDR IR - add xdr-get-endpoints-by-status command

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39923

## Description
- Added the ***xdr-get-endpoints-by-status*** command to XDR IR integration to gets the `total_count` value from `/endpoints/get_endpoint/` response.
- Use the ***xdr-get-endpoints-by-status*** command output in `XDRDisconnectedEndpoints`,`XDRConnectedEndpoints` scripts instead of ***xdr-get-endpoints***

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CortexXDR/.pack-ignore [hunk 1]
```diff
@@ -41,3 +41,9 @@ ignore=IF100
 
 [file:widget-Cortex_XDR_Disconnected_Endpoints.json]
 ignore=BA113
+
+[file:XDRDisconnectedEndpoints.yml]
+ignore=BC103
+
+[file:XDRConnectedEndpoints.yml]
+ignore=BC103
```
Comment (yaakovi) [inline Packs/CortexXDR/.pack-ignore:49]: you can't ignore those

File: Packs/CortexXDR/ReleaseNotes/3_0_27.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+- Added the ***xdr-get-endpoints-by-status*** command.
+- Updated the Docker image to: *demisto/python3:3.9.6.24019*.
+
+#### Scripts
+##### XDRDisconnectedEndpoints
+- use the ***xdr-get-endpoints-by-status*** command to gets the endpoint count.
+- Updated the Docker image to: *demisto/python3:3.9.6.24019*.
+##### XDRConnectedEndpoints
+- use the ***xdr-get-endpoints-by-status*** command to gets the endpoint count.
+- Updated the Docker image to: *demisto/python3:3.9.6.24019*.
```
Comment (yaakovi) [inline Packs/CortexXDR/ReleaseNotes/3_0_27.md:13]:
```suggestion
#### Scripts
##### XDRDisconnectedEndpoints
- Uses the ***xdr-get-endpoints-by-status*** command to get the disconnected endpoint count.
- Updated the Docker image to: *demisto/python3:3.9.6.24019*.

##### XDRConnectedEndpoints
- Uses the ***xdr-get-endpoints-by-status*** command to get the connected endpoint count.
- Updated the Docker image to: *demisto/python3:3.9.6.24019*.
```

File: Packs/CortexXDR/Scripts/XDRConnectedEndpoints/XDRConnectedEndpoints.yml [hunk 3]
```diff
@@ -16,13 +16,10 @@ args:
   description: 'The end date by which to filter incidents. Date format will be the
       same as in the incidents query page, for example: "3 days ago", "2019-01-01T00:00:00
       +0200").'
-- name: limit
```
Comment (yaakovi) [inline Packs/CortexXDR/Scripts/XDRConnectedEndpoints/XDRConnectedEndpoints.yml:19]: why did you remove the limit? please restore

File: Packs/CortexXDR/Scripts/XDRDisconnectedEndpoints/XDRDisconnectedEndpoints.py [hunk 4]
```diff
@@ -9,27 +9,23 @@ def main():
...
-        disconnected_endpoints = [endpoint for endpoint in endpoints if endpoint.get('endpoint_status') != 'CONNECTED']
-        disconnected_endpoints_num = len(disconnected_endpoints)
-
-        return_results(str(disconnected_endpoints_num))
+        count = list(res[0].get('EntryContext').values())[0].get('count')
+        return_results(count)
     except Exception as ex:
         demisto.error(traceback.format_exc())  # print the traceback
-        return_error(f'Failed to execute XDRDisconnectedEndpoints. Error: {str(ex)}')
+        return_error(f'Failed to execute XDRConnectedEndpoints. Error: {str(ex)}')
```
Comment (yaakovi) [inline Packs/CortexXDR/Scripts/XDRDisconnectedEndpoints/XDRDisconnectedEndpoints.py:28]: revert

File: Packs/CortexXDR/Scripts/XDRDisconnectedEndpoints/XDRDisconnectedEndpoints.yml [hunk 5]
```diff
@@ -16,13 +16,10 @@ args:
   description: 'The end date by which to filter incidents. Date format will be the
       same as in the incidents query page, for example: "3 days ago", "2019-01-01T00:00:00
       +0200").'
-- name: limit
```
Comment (yaakovi) [inline Packs/CortexXDR/Scripts/XDRDisconnectedEndpoints/XDRDisconnectedEndpoints.yml:19]: same here

File: Packs/CortexXDR/pack_metadata.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
     "name": "Palo Alto Networks Cortex XDR - Investigation and Response",
     "description": "Automates Cortex XDR incident response, and includes custom Cortex XDR incident views and layouts to aid analyst investigations.",
     "support": "xsoar",
-    "currentVersion": "3.0.26",
+    "currentVersion": "3.0.27",
```
Comment (yaakovi) [inline Packs/CortexXDR/pack_metadata.json:5]: switch to minor version

**Review by yaakovi — CHANGES_REQUESTED** [review]

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 7]
```diff
@@ -2891,3 +2891,48 @@ There is no context output for this command.
...
+            "count": 4,
+            "status": "connected"
+        }
+    }
+}
+```
+
+#### Human Readable Output
+
+>connected endpoins count: 4
```
Comment (yaakovi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:2938]:
```suggestion
>connected endpoints count: 4
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 8]
```diff
@@ -3208,6 +3241,31 @@ def run_script_kill_process_command(client: Client, args: Dict) -> List[CommandR
...
+        arg=args.get('last_seen_lte'),
+        arg_name='last_seen_lte'
+    )
+
+    endpoints_count, raw_res = client.get_endpoints_by_status(status, last_seen_gte=last_seen_gte, last_seen_lte=last_seen_lte)
+
+    ec = {'status': status, 'count': endpoints_count}
+
+    return CommandResults(
+        readable_output=f'{status} endpoins count: {endpoints_count}',
```
Comment (yaakovi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:3262]:
```suggestion
        readable_output=f'{status} endpoints count: {endpoints_count}',
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**